images/virtualization-artifact/pkg/controller/vmip/internal/state/state.go

```diff
@@ -70,6 +71,20 @@ func (s *state) VirtualMachineIPLease(ctx context.Context) (*virtv2.VirtualMachi
 		}
 	}
 
+	if s.lease == nil {
+		var leases virtv2.VirtualMachineIPAddressLeaseList
+		err = s.client.List(ctx, &leases, &client.MatchingFields{
+			indexer.IndexFieldVMIPLeaseByVMIP: s.vmip.Name,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		if len(leases.Items) > 0 {
+			s.lease = &leases.Items[0]
```

Comment:
we can find several leases. Maybe we'll choose the one in the bound phase?

Reply:
Ok. I added watch conditions.